contracts/SiphonDepositor.sol

```diff
@@ -0,0 +1,177 @@
+}
+
+contract SiphonDepositor is Ownable {
+    using AuraMath for uint256;
+    using SafeERC20 for IERC20;
+    using SafeERC20 for ICvx;
+
+    IERC20 public immutable lpToken;
+    IERC20 public immutable crv;
+    IBooster public immutable booster;
+    ICvx public immutable cvx;
+    IrCvx public immutable rCvx;
+    IAuraLocker public immutable auraLocker;
+    uint256 public immutable pid;
+
+    /**
+     * @dev Penalty basis points 250 == 25%
+     */
+    uint256 public immutable penaltyBp;
+
+    /**
+     * @dev total amount of farmed CVX (AURA) tokens
+     */
+    uint256 public farmedTotal;
+
+    constructor(
+        IERC20 _lpToken,
+        IERC20 _crv,
+        IBooster _booster,
+        ICvx _cvx,
+        IrCvx _rCvx,
+        IAuraLocker _auraLocker,
+        uint256 _pid,
+        uint256 _penaltyBp
+    ) {
+        lpToken = _lpToken;
+        crv = _crv;
+        booster = _booster;
+        cvx = _cvx;
+        rCvx = _rCvx;
+        auraLocker = _auraLocker;
+        pid = _pid;
+        penaltyBp = _penaltyBp;
+    }
+
+    /**
+     * @dev deposit "lpTokens" into the booster
+     */
+    function deposit() external {
+        lpToken.approve(address(booster), type(uint256).max);
+        uint256 bal = lpToken.balanceOf(address(this));
+        booster.deposit(pid, bal, true);
+    }
+
+    /**
+     * @dev withdraw "lpTokens" from the booster
+     */
+    function withdraw(uint256 _amount) external {
+        booster.withdraw(pid, _amount);
+    }
+
+    /**
+     * @dev Siphon AURA tokens by depositing BAL into the Booster
+     *      and then calling earmark rewards which will send the BAL
+     *      to the siphon pools BaseRewardPool
+     */
+    function siphon() external {
+        uint256 bal = crv.balanceOf(address(this));
+        crv.transfer(address(booster), bal);
+        // Mint rCvx at a rate of 1:1 rAURA:BALRewards
+        rCvx.mint(address(this), bal);
+        booster.earmarkRewards(pid);
+    }
+
+    /**
+     * @dev Call getReward on the BaseRewardPool which will return the
+     *      BAL we previously depoisted minus the incentives that were paid
+     *      Along with a pro rata amount of AURA tokens
+     */
+    function getReward() external {
+        uint256 balBefore = cvx.balanceOf(address(this));
+        (, , , address crvRewards, , ) = booster.poolInfo(pid);
+        IBaseRewardPool(crvRewards).getReward(address(this), false);
+        uint256 balAfter = cvx.balanceOf(address(this));
+        // Increment total farmed amount
+        farmedTotal += (balAfter - balBefore);
+    }
+
+    /**
+     * @dev How much AURA you will get for rAURA
+     *      rAURA can be redeemed at a rate determined by the totalSupply
+     *      of rAURA and totalAmount of AURA the depositor has farmed.
+     */
+    function getAmountOut(uint256 _amount) public view returns (uint256) {
+        uint256 totalSupply = rCvx.totalSupply();
+        return (_amount * farmedTotal) / totalSupply;
```

Comment:
We actually don't need to track farmedTotal we just use `cvx.balanceOf(address(this))`

Reply:
Yep